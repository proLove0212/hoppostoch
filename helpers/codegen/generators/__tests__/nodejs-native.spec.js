import { NodeJsNativeCodegen } from "../nodejs-native"
import {
  TEST_BEARER_TOKEN,
  TEST_HEADERS,
  TEST_HTTP_PASSWORD,
  TEST_HTTP_USER,
  TEST_PATH_NAME,
  TEST_QUERY_STRING,
  TEST_RAW_PARAMS_JSON,
  TEST_RAW_PARAMS_XML,
  TEST_RAW_REQUEST_BODY,
  TEST_URL,
} from "../__fixtures__/test-data"

describe("generate request for NodeJs Native", () => {
  test("generate GET request", () => {
    const result = NodeJsNativeCodegen.generator({
      url: TEST_URL,
      pathName: TEST_PATH_NAME,
      queryString: TEST_QUERY_STRING,
      auth: "Basic Auth",
      httpUser: TEST_HTTP_USER,
      httpPassword: TEST_HTTP_PASSWORD,
      method: "GET",
      headers: TEST_HEADERS,
    })
    expect(result).toMatchSnapshot()
  })

  test("generate POST request for JSON", () => {
    const result = NodeJsNativeCodegen.generator({
      url: TEST_URL,
      pathName: TEST_PATH_NAME,
      queryString: TEST_QUERY_STRING,
      auth: "Bearer Token",
      bearerToken: TEST_BEARER_TOKEN,
      method: "POST",
      rawInput: true,
      rawParams: TEST_RAW_PARAMS_JSON,
      contentType: "application/json",
      headers: TEST_HEADERS,
    })
    expect(result).toMatchSnapshot()
  })

  test("generate POST request for XML", () => {
    const result = NodeJsNativeCodegen.generator({
      url: TEST_URL,
      pathName: TEST_PATH_NAME,
      queryString: TEST_QUERY_STRING,
      auth: "OAuth 2.0",
      bearerToken: TEST_BEARER_TOKEN,
      method: "POST",
      rawInput: true,
      rawParams: TEST_RAW_PARAMS_XML,
      contentType: "application/xml",
      headers: TEST_HEADERS,
    })
    expect(result).toMatchSnapshot()
  })

  test("generate PUT request for www-form-urlencoded", () => {
    const result = NodeJsNativeCodegen.generator({
      url: TEST_URL,
      pathName: TEST_PATH_NAME,
      queryString: TEST_QUERY_STRING,
      method: "PUT",
      rawInput: false,
      rawRequestBody: TEST_RAW_REQUEST_BODY,
      contentType: "application/x-www-form-urlencoded",
    })
    expect(result).toMatchSnapshot()
  })
})
