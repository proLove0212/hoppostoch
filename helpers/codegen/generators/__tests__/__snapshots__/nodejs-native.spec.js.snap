// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate request for NodeJs Native generate GET request 1`] = `
"const http = require('http');

const url = 'https://httpbin.org/path/to?a=b';
const options = {
  method: 'GET',
  headers: {
    \\"Authorization\\": \\"Basic bW9ja1VzZXI6bW9ja1Bhc3N3b3Jk\\",
    \\"h1\\": \\"h1v\\",
    \\"h2\\": \\"h2v\\"
  }};

const request = http.request(url, options, (response) => {
  console.log(response);
});

request.on('error', (e) => {
  console.error(e);
});
request.end();"
`;

exports[`generate request for NodeJs Native generate POST request for JSON 1`] = `
"const http = require('http');

const url = 'https://httpbin.org/path/to?a=b';
const options = {
  method: 'POST',
  headers: {
    \\"Authorization\\": \\"Bearer abcdefghijklmn\\",
    \\"Content-Type\\": \\"application/json; charset=utf-8\\",
    \\"h1\\": \\"h1v\\",
    \\"h2\\": \\"h2v\\"
  }};

const request = http.request(url, options, (response) => {
  console.log(response);
});

request.on('error', (e) => {
  console.error(e);
});

request.write(JSON.stringify({\\"foo\\": \\"bar\\", \\"baz\\": \\"qux\\"}));
request.end();"
`;

exports[`generate request for NodeJs Native generate POST request for XML 1`] = `
"const http = require('http');

const url = 'https://httpbin.org/path/to?a=b';
const options = {
  method: 'POST',
  headers: {
    \\"Authorization\\": \\"Bearer abcdefghijklmn\\",
    \\"Content-Type\\": \\"application/xml; charset=utf-8\\",
    \\"h1\\": \\"h1v\\",
    \\"h2\\": \\"h2v\\"
  }};

const request = http.request(url, options, (response) => {
  console.log(response);
});

request.on('error', (e) => {
  console.error(e);
});

request.write(\`<?xml version='1.0' encoding='utf-8'?>
<xml>
  <element foo=\\"bar\\"></element>
</xml>\`);
request.end();"
`;

exports[`generate request for NodeJs Native generate PUT request for www-form-urlencoded 1`] = `
"const http = require('http');

const url = 'https://httpbin.org/path/to?a=b';
const options = {
  method: 'PUT',
  headers: {
    \\"Content-Type\\": \\"application/x-www-form-urlencoded; charset=utf-8\\"
  }};

const request = http.request(url, options, (response) => {
  console.log(response);
});

request.on('error', (e) => {
  console.error(e);
});

request.write(\`foo=bar&baz=qux\`);
request.end();"
`;
